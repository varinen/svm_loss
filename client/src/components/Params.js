import React, {Component} from 'react';
import {connect} from 'react-redux';
import MathJax from 'react-mathjax';
import Button from 'react-bootstrap/Button';
import {getParams} from '../actions/params';
import ParamControl from  './ParamControl';

const formulas = {
    f1: `W`,
    f2: `b`
};

export class Params extends Component {

    getParams = () => {
        this.props.getParams(1)
    };

    componentDidMount() {
        this.props.getParams();
    }

    render() {
        return (
            <MathJax.Provider>
                <div className="params card shadow">
                    <div className="card-header py-3">
                        <h6 className="m-0 font-weight-bold text-primary">
                            Classifier Parameters</h6>
                    </div>
                    <div className="card-body">
                        <article>
                            <p>
                                Parameters <MathJax.Node inline
                                                         formula={formulas.f1}/>
                                , <MathJax.Node inline
                                                formula={formulas.f2}/> are
                                shown below. The value is in <b>bold</b> and
                                its gradient
                                (computed with backprop) is in <span
                                className="text-red font-italic">red, italic</span> below. Click the triangles to
                                control the parameters.
                            </p>
                            <ParamControl />
                            <div>
                                <Button className="btn btn-primary randomize-params"
                                        onClick={this.getParams}>
                                    Randomize parameters
                                </Button>
                            </div>


                        </article>
                    </div>
                </div>
            </MathJax.Provider>

        );
    }
}
const mapPropsToStore = state => {
    const {params} = state;
    return params
};
export default connect(mapPropsToStore, {getParams})(Params);