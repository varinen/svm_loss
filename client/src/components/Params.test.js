import React from 'react';
import {shallow} from 'enzyme';
import Params from './Params';

describe('Params', () => {
    const params = shallow(<Params/>);

    it('renders correctly', () => {
        expect(params).toMatchSnapshot();
    });
});

