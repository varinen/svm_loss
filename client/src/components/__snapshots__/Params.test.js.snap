// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Params renders correctly 1`] = `
<MathJaxProvider
  options={
    Object {
      "showMathMenu": false,
      "showMathMenuMSIE": false,
      "tex2jax": Object {
        "inlineMath": Array [],
      },
    }
  }
  script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
>
  <div
    className="params card shadow"
  >
    <div
      className="card-header py-3"
    >
      <h6
        className="m-0 font-weight-bold text-primary"
      >
        Classifier Parameters
      </h6>
    </div>
    <div
      className="card-body"
    >
      <article>
        <p>
          Parameters 
          <MathJaxNode
            formula="W"
            inline={true}
          />
          , 
          <MathJaxNode
            formula="b"
            inline={true}
          />
           are shown below. The value is in 
          <b>
            bold
          </b>
           and its gradient (computed with backprop) is in 
          <span
            className="text-red font-italic"
          >
            red, italic
          </span>
          below. Click the triangles to control the parameters.
        </p>
        <div>
          <Button
            active={false}
            className="btn btn-primary"
            disabled={false}
            type="button"
            variant="primary"
          >
            Randomize parameters
          </Button>
        </div>
      </article>
    </div>
  </div>
</MathJaxProvider>
`;
